### PR TITLE
Resolve protocol http, https when not in lowercase

### LINKED ISSUE #21395: Android - Linking won't openURL with uppercase in protocol http/https
- [v] Review the documentation: https://facebook.github.io/react-native
  - [v] Search for existing issues: https://github.com/facebook/react-native/issues
  - [not sure] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment
Environment:
  OS: macOS High Sierra 10.13.6
  Node: 8.12.0
  Yarn: 1.10.1
  npm: 6.4.1
  Watchman: Not Found
  Xcode: Xcode 10.0 Build version 10A255
  Android Studio: 3.1 AI-173.4907809

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.53.0 => 0.53.0

Run `react-native info` in your terminal and paste its contents here.
https://snack.expo.io/@hyunjongl/android-linking-protocol-problem  
Expo can be used to reproduce.

## Description
Describe your issue in detail. Include screenshots if needed. If this is a regression, let us know.
https://snack.expo.io/@hyunjongl/android-linking-protocol-problem  
The Android Intent can't open URL with uppercase in http/https. However iOS can open it.

## Reproducible Demo
https://snack.expo.io/@hyunjongl/android-linking-protocol-problem  
Run with Android. You won't open urls with uppercase in the protocol of http/https.

Let us know how to reproduce the issue. Include a code sample, share a project, or     share an app that reproduces the issue using https://snack.expo.io/. Please follow the guidelines for providing a MCVE: https://stackoverflow.com/help/mcve

### PR DESCRIPTION
**Pull request by hyunjongL**
Thank you for sending the PR! We appreciate you spending the time to work on these changes.
Help us understand your motivation by explaining why you decided to make this change.

If this PR fixes an issue, type "Fixes #issueNumber" to automatically close the issue when the PR is merged.

_Pull requests that expand test coverage are more likely to get reviewed. Add a test case whenever possible!_

Test Plan:
----------
Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!  

I have added to my react native app that has to open URLs with protocols(Https) that have uppercases. Added console log of the url back and forth to test if the url has changed.
You can easily build an react-native app like https://snack.expo.io/@hyunjongl/android-linking-protocol-problem and see if this resolves it.
image
This is the screenshot of the console log where I try to open a URL with Https, and the code will automatically change it into https.

Release Notes:
--------------
Fixes #21395
[ANDROID][ENHANCEMENT][Libraries/Linking/Linking.js]
https://snack.expo.io/@hyunjongl/android-linking-protocol-problem   
  
There issue is that when you have a URL with protocol that has an uppercase, it works in iOS, but won't work in Android. (check the Expo above).  
There are a lot of user generated URLs in social platforms, and I bet it is not a good Idea to change what the user have wrote. Also currently I feel like iOS and Android behave differently in this issue which makes me feel bad.  
So, I am suggesting a way to change the protocol['http://', 'https://'] into lowercase just before opening the browser. However, as I am very new to this community, it might not be the best place to add the lines, so comments are very welcome.   

TL;DR Added few lines to change the protocol string to lowercase when some are in uppercase.

### CODE REVIEW

File: Libraries/Linking/Linking.js [hunk 1]
```diff
@@ -57,6 +57,11 @@ class Linking extends NativeEventEmitter {
    * See https://facebook.github.io/react-native/docs/linking.html#openurl
    */
   openURL(url: string): Promise<any> {
+    if (url.toLowerCase().indexOf('https://') == 0) {
```
[the remark this answers is not in the document]
Comment (hyunjongL) [reply Libraries/Linking/Linking.js]: okay ^_^

File: Libraries/Linking/Linking.js [hunk 2]
```diff
@@ -57,6 +57,11 @@ class Linking extends NativeEventEmitter {
...
+      url = url.replace(url.substr(0, 8), 'https://');
+    } else if (url.toLowerCase().indexOf('http://') == 0) {
```
[the remark this answers is not in the document]
Comment (hyunjongL) [reply Libraries/Linking/Linking.js]: Yes, master

**Comment by hyunjongL** [comment]
Can someone help me understand why one of the tests have failed?

File: Libraries/Linking/Linking.js [hunk 3]
```diff
@@ -57,6 +57,11 @@ class Linking extends NativeEventEmitter {
    * See https://facebook.github.io/react-native/docs/linking.html#openurl
    */
   openURL(url: string): Promise<any> {
+    if (url.toLowerCase().indexOf('https://') === 0) {
```
Comment (RSNara) [inline Libraries/Linking/Linking.js]: nit: It'd be nice to have a comment to clarify why this needs to be done. Something like:
```JavaScript
// Android requires url protocols to be lowercase (i.e: Https:// doesn't work, but https:// does)
```

**Comment by RSNara** [comment]
@hyunjongL I don't think your PR broke CI. So, I wouldn't worry about it.

**Comment by hyunjongL** [comment]
@RSNara check out the changes!

**Comment by RSNara** [comment]
Thanks for this PR!
